### PR TITLE
Add a generic log retrieval client to Armada Python Client

### PR DESCRIPTION
**Pull request by ClifHouck**
Adds a job log retrieval client to Armada's Python client library. Users of the library are meant to use `JobLogClient` while underneath the hood it relies on a new `BinocularsClient`.

### CODE REVIEW

**Review by d80tb7 — DISMISSED** [review]

File: client/python/armada_client/internal/binoculars_client.py
Comment (dejanzele) [inline client/python/armada_client/internal/binoculars_client.py:13]: Should it by default expect a secure connection?

Comment (ClifHouck) [reply client/python/armada_client/internal/binoculars_client.py:13]: The way it's written, by default it will try to use SSL.

**Comment by ClifHouck** [comment]
I think this is good to go, if I can get approvals again.

**Review by dejanzele — APPROVED** [review]

**Review by d80tb7 — APPROVED** [review]

**Pull request merged**
